packages/cysync-core/src/pages/OnBoarding/DeviceDetection/index.tsx

```diff
@@ -35,14 +36,37 @@ const DeviceNotConnectedDialogBox: React.FC<{
   </DialogBox>
 );
 
+const OnboardingMap: Record<OnboardingStep, string> = {
+  [OnboardingStep.ONBOARDING_STEP_VIRGIN_DEVICE]:
+    routes.onboarding.deviceAuthentication.path,
+  [OnboardingStep.ONBOARDING_STEP_DEVICE_AUTH]:
+    routes.onboarding.joystickTraining.path,
+  [OnboardingStep.ONBOARDING_STEP_JOYSTICK_TRAINING]:
+    routes.onboarding.cardTraining.path,
+  [OnboardingStep.ONBOARDING_STEP_CARD_CHECKUP]:
+    routes.onboarding.cardAuthentication.path,
+  [OnboardingStep.ONBOARDING_STEP_CARD_AUTHENTICATION]:
+    routes.onboarding.congratulations.path,
+  [OnboardingStep.ONBOARDING_STEP_COMPLETE]:
+    routes.onboarding.congratulations.path,
+  [OnboardingStep.UNRECOGNIZED]: routes.onboarding.deviceAuthentication.path,
+};
+
 export const DeviceDetection: React.FC = () => {
-  const { connection } = useDevice();
+  const { connection, connectDevice } = useDevice();
   const navigateTo = useNavigateTo();
   const lang = useAppSelector(selectLanguage);
 
+  const gotoNextPage = async () => {
+    if (!connection) return;
+    const app = await ManagerApp.create(await connectDevice(connection.device));
+    const res = (await app.getDeviceInfo()).onboardingStep;
+    console.log({ res });
+    navigateTo(OnboardingMap[res]);
+  };
```

Comment:
We already get the device info when the device is connected, we don't need to fetch it again.